packages/tailwindcss/src/variants.ts

```diff
@@ -711,6 +711,18 @@ export function createVariants(theme: Theme): Variants {
 
   staticVariant('inert', ['&:is([inert], [inert] *)'])
 
+  variants.functional('in', (ruleNode, variant) => {
+    if (!variant.value || variant.modifier) return null
+
+    // Named values should be alphanumeric. This prevents `in-foo-bar` from
+    // being used as a variant.
+    if (variant.value.kind === 'named' && !/^[a-zA-Z]+$/i.test(variant.value.value)) {
```

Comment:
We should lift the regex out of the function to ensure its only compiled once

Reply:
Did some benchmarking and a manual loop is faster. So used that instead (also pushed the benchmark file).

Reply:
If we make it a compound variant and only allow `in-<other-variant>` like `in-data-visible` or `in-[p]` then we don't need the `isAlpha` check because `in-p` requires `p` to be a variant.